pyfar/dsp/dsp.py

```diff
@@ -850,3 +851,57 @@ def _cross_fade(first, second, indices):
     result = first * window_first + second * window_second
 
     return result
+
+
+def time_shift(signal, shift, unit='samples'):
+    """Apply a time-shift to a signal.
+
+    The shift is performed as a cyclic shift on the time axis, potentially
+    resulting in non-causal signals for negative shift values.
+
+    Parameters
+    ----------
+    signal : Signal
+        The signal to be shifted
+    shift : int, float
+        The time-shift value. A positive value will result in right shift on
+        the time axis (delaying of the signal), whereas a negative value
+        yields a left shift on the time axis (non-causal shift to a earlier
+        time).
+    unit : str, optional
+        Unit of the shift variable, this can be either samples or seconds.
```

Comment:
To be consistent with units in other functions, I'd suggest to have `'samples'` and `'s` instead of `'seconds'`, also `'ms'` would be nice :)
```suggestion
        Unit of the shift variable, this can be either ``'samples'``,  ``'s'`` for seconds or ``'ms'`` for milliseconds.
```

Reply:
Frankly I don't really see the need to specify milliseconds, since it's not really more convenient than adding a `e-3` to the specified value. And I don't like the inconsistency in spelling samples, but abbreviating seconds to s either. But I'd let myself get overruled here.